### PR TITLE
Add SBB app

### PR DESCRIPTION
**Pull request by LukiLeu**
This app generates a real-time timetable for any train station in the Swiss Public Transportation Network. It shows the departure time and destination of the next five trains that are leaving the chosen station. If a train is delayed, this delay is also shown.

The station can be set with the _station_ parameter. Additionally, with the parameter _skiptime_, it is possible to show departures in the future. As an example: If this parameter is set to _10_ Minutes, the departures between _now_ and _now+10 minutes_ are skipped and the next five departures departing after _now+10 minutes_ are shown.

sbb

### CODE REVIEW

**Comment by LukiLeu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by LukiLeu** [comment]
Why is the buildifier failing here? I don't see the reason.

**Comment by henryso** [comment]
You can click on the Details links to see the errors.

The first thing you should do is merge your repository's main branch with the tidbyt repo's main branch.  This should fix the build-and-test failure.

The second thing you should do is run `make lint` which should fix the lint errors, or you can alternately change your file manually.

**Comment by LukiLeu** [comment]
> You can click on the Details links to see the errors.
> 
> The second thing you should do is run `make lint` which should fix the lint errors, or you can alternately change your file manually.

I did run `make lint` on my ubuntu machine and it did not report any error. If I look into the details here, it just reports `Error: Process completed with exit code 4.` but I can't find any additional description of where the exact error is.

**Comment by LukiLeu** [comment]
Ah I see `make lint` does change my file without reporting what might be wrong.

**Comment by henryso** [comment]
Yeah, it's a little misleading because lint is typically a "check my code" thing rather that a "check and fix" thing.

File: apps/sbbtimetable/sbb_timetable.star [hunk 1]
```diff
@@ -0,0 +1,203 @@
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+load("time.star", "time")
+load("schema.star", "schema")
+
+# Define some constants
+SBB_URL = "https://fahrplan.search.ch/api/stationboard.json?show_delays=1&transportation_types=train&stop="
```
Comment (betterengineering) [inline apps/sbbtimetable/sbb_timetable.star:10]: Notes: 
- API Docs: https://fahrplan.search.ch/api/help
- Certificate is from R3 and will expire in April. We've had some expire like, two days after we land a new app and cause all sorts of problems 🤦 
- Looks like we'll be rate limited at  10080 departure/arrival tables
- Seems like it's fine to submit queries as long as we are under the req/s limit and I don't see anything that says we need attribution??

File: apps/sbbtimetable/sbb_timetable.star [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "station",
+                name = "Station",
+                desc = "Station from which the timetable shall be shown.",
+                icon = "train",
+            ),
+            schema.Text(
+                id = "skiptime",
```
Comment (betterengineering) [inline apps/sbbtimetable/sbb_timetable.star:197]: Fancy! Effectively, how far away do you live from the train??

File: apps/sbbtimetable/sbb_timetable.star [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        # Get the cached response
+        print("Hit! Displaying cached data.")
+        resp = json.decode(resp_cached)
+    else:
+        # Get a new reponse
+        print("Miss! Calling API.")
+        resp = http.get("%s%s" % (SBB_URL, station))
+        if resp.status_code != 200:
+            fail("Request failed with status %d", resp.status_code)
+        cache.set("sbb_%s" % station, resp.body(), ttl_seconds = 120)
```
Comment (betterengineering) [inline apps/sbbtimetable/sbb_timetable.star:51]: Question: Any idea how many stations there are? We'll hit this API potentially `(24 hours * 60 minutes) / 2 minutes = 720` times the number of stations if this app is popular. At the `10080` limit we can only support 14 stations before we'd need an API key. Seems fine for now, but we'll want to keep an eye on it.

**Review by betterengineering — APPROVED** [review]
This looks good to me! I'm going to get this landed so we can start trying it out internally. If it still looks good, I'll launch it to the broader user base. Feel free to reply to my comments and I'll circle back tomorrow!

File: apps/sbbtimetable/sbb_timetable.star (hunk 1 above)
Comment (LukiLeu) [reply apps/sbbtimetable/sbb_timetable.star:10]: I can't find anything that says we need attribution: [https://fahrplan.search.ch/api/terms.en.html](https://fahrplan.search.ch/api/terms.en.html)

File: apps/sbbtimetable/sbb_timetable.star (hunk 2 above)
Comment (LukiLeu) [reply apps/sbbtimetable/sbb_timetable.star:197]: Actually, I have a train station right next to my home. But on weekdays I use a train station to commute that is a 30 minutes car drive away.

File: apps/sbbtimetable/sbb_timetable.star (hunk 3 above)
Comment (LukiLeu) [reply apps/sbbtimetable/sbb_timetable.star:51]: I think at the moment there are somewhat over 1000 stations (only train). If, later also bus and ship is added, this number will go up to ~29'000 stations.
It probably depends how many tidbyt users you have in Switzerland :-). But if you hit the rate limit, it should also be possible to increase the ttl value to five minutes.

**Pull request merged**